ethportal-peertest/src/scenarios/find.rs

```diff
@@ -1,61 +1,106 @@
-use serde_json::{json, Value};
+use std::ops::Deref;
 
 use crate::{
     jsonrpc::{make_ipc_request, JsonRpcRequest, HISTORY_CONTENT_VALUE},
     Peertest, PeertestConfig,
 };
-use trin_types::{constants::CONTENT_ABSENT, jsonrpc::params::Params};
-
-pub fn test_trace_recursive_find_content(_peertest_config: PeertestConfig, peertest: &Peertest) {
-    let uniq_content_key = "0x0015b11b918355b1ef9c5db810302ebad0bf2544255b530cdce90674d5887bb286";
-    // Store content to offer in the testnode db
-    let store_request = JsonRpcRequest {
-        method: "portal_historyStore".to_string(),
-        id: 11,
-        params: Params::Array(vec![
-            Value::String(uniq_content_key.to_string()),
-            Value::String(HISTORY_CONTENT_VALUE.to_string()),
-        ]),
-    };
+use ethportal_api::{
+    types::portal::TraceContentInfo, ContentValue, HistoryContentKey, HistoryNetworkApiClient,
+    PossibleHistoryContentValue,
+};
+use serde_json::json;
+use trin_types::content_value::HistoryContentValue;
+use trin_types::discv5::NodeId;
+use trin_types::jsonrpc::params::Params;
+use trin_utils::bytes::hex_decode;
+
+pub async fn test_trace_recursive_find_content(
+    _peertest_config: PeertestConfig,
+    peertest: &Peertest,
+) {
+    let node_a_client = reth_ipc::client::IpcClientBuilder::default()
+        .build(&peertest.bootnode.web3_ipc_path)
+        .await
+        .unwrap();
 
-    let store_result = make_ipc_request(&peertest.bootnode.web3_ipc_path, &store_request).unwrap();
-    assert!(store_result.as_bool().unwrap());
+    let node_b_client = reth_ipc::client::IpcClientBuilder::default()
+        .build(&peertest.nodes[0].web3_ipc_path)
+        .await
+        .unwrap();
 
-    // Send trace recursive find content request
-    let request = JsonRpcRequest {
-        method: "portal_historyTraceRecursiveFindContent".to_string(),
-        id: 12,
-        params: Params::Array(vec![json!(uniq_content_key)]),
-    };
+    let uniq_content_key =
+        "\"0x0015b11b918355b1ef9c5db810302ebad0bf2544255b530cdce90674d5887bb286\"";
+    let history_content_key: HistoryContentKey = serde_json::from_str(uniq_content_key).unwrap();
```

Comment:
This feels like it has extra steps that I don't understand. Why go into a json-encoded string and back out?

I see some other places that do this same concept like this, like:
```rs
    let content_key: HistoryContentKey =
        serde_json::from_value(json!(HISTORY_CONTENT_KEY)).unwrap();
```
which is a little better. But still, it seems like going in and out of json doesn't really do anything for us.

Something equivalent to this seems like the direct path that doesn't involve unnecessary json:
```rs
let key_bytes = "0015b11b918355b1ef9c5db810302ebad0bf2544255b530cdce90674d5887bb286";
let history_content_key = HistoryContentKey::from_bytes(hex::decode(key_bytes));
```